app/models/product_import/entry_validator.rb

```diff
@@ -79,10 +79,9 @@ def mark_as_new_variant(entry, product_id)
         if entry.attributes['on_hand'].present?
           new_variant.on_hand = entry.attributes['on_hand']
         end
+        check_on_hand_nil(entry, new_variant)
       end
 
-      check_on_hand_nil(entry, new_variant)
```

Comment:
- If the sheet doesn't have the `units` present, then the variant is not saved due to model validation
- After that, while assigning `on_hand` value in this method, the error is raised that the variant is not created first
- So we need to make sure that the variant is created before calling this method